### ISSUE TITLE
Wrong Text extracted in 'complex' table

### ISSUE THREAD

**Issue by erwanlr** [yob/pdf-reader#110]
When using a complex table (multi rows, columns), the text extracted is wrong: some letters are duplicated, others moved etc

Code used:

``` ruby
# coding: utf-8

require 'rubygems'
require 'pdf/reader'

reader = PDF::Reader.new('sample.pdf')
puts reader.page(1).text
```

Output:

```
Title:    Testing the title

Data


Tite2:e    AnnotherLoong TestNaame

Data
```

![sample](https://f.cloud.github.com/assets/1668843/1138387/0c370636-1c75-11e3-9104-5636f64c3095.png)

I can provide the sample.pdf file via email if needed :)

**Comment by AsrorWali** [comment]
I have tried what @gpinkham said but that didnt work me. I looked through the code and the problem was completly elsewhere for me. It said that @state.font_size was 300

So in text_run  if (other.x - endx) <( font_size * 0.2) would be < 60 which caused the problems for me. When I replaced @state.font_size with 12, I got the desired results.

I havent figured out why he gives me font_size 300, gonna try to look for it

**Comment by yob** [comment]
Without a sample file it's hard to be sure, however there's a very high chance the positioning issues described here are addressed by the fixes I'm working on for #397. Hopefully I'll land them on main in the next few days and we can resolve this issue at the same time
